pages/price-feeds/use-real-time-data/starknet.mdx

```diff
@@ -0,0 +1,210 @@
+1. Call `pyth.get_price_no_older_than` to read the price, providing the [price feed ID](https://pyth.network/developers/price-feed-ids) you wish to read.
+
+The code snippet below provides an example of how to update the Pyth price feeds on Starknet using the `pyth-starknet-js` in JavaScript:
+
+```ts {52} copy
+// `pyth-starknet-js` is intended to be combined with `price-service-client` and `starknet-js`.
+import { Account, Contract, RpcProvider, shortString } from "starknet";
+import { PriceServiceConnection } from "@pythnetwork/price-service-client";
+import {
+  ByteBuffer,
+  ERC20_ABI,
+  ETH_TOKEN_ADDRESS,
+  PYTH_ABI,
+  PYTH_CONTRACT_ADDRESS_SEPOLIA,
+} from "@pythnetwork/pyth-starknet-js";
+
+// Create a `RpcProvider` instance to interact with Starknet RPC.
+const provider = new RpcProvider({
+  nodeUrl: "https://starknet-sepolia.public.blastapi.io/rpc/v0_6",
+});
+
+// Create a `Contract` instance to interact with a fee token contract on Starknet
+// You can use STRK or ETH to pay fees, but using STRK is recommended.
+const strkErc0Contract = new Contract(ERC20_ABI, STRK_TOKEN_ADDRESS, provider);
+
+// Create a `Contract` instance to interact with the Pyth contract on Starknet.
+const pythContract = new Contract(
+  PYTH_ABI,
+  PYTH_CONTRACT_ADDRESS_SEPOLIA,
+  provider
+);
+
+// Import your account data from environment variables.
+// You'll need to set them before running the code.
+const privateKey0 = process.env.ACCOUNT_PRIVATE_KEY;
+if (privateKey0 === undefined) {
+  throw new Error("missing ACCOUNT_PRIVATE_KEY");
+}
+const account0Address = process.env.ACCOUNT_ADDRESS;
+if (account0Address === undefined) {
+  throw new Error("missing ACCOUNT_ADDRESS");
+}
+const account0 = new Account(provider, account0Address, privateKey0);
+
+// Note: Get the Stable Hermes service URL from https://docs.pyth.network/price-feeds/api-instances-and-providers/hermes
+const connection = new PriceServiceConnection("https://hermes.pyth.network", {
+  priceFeedRequestConfig: {
+    binary: true,
+  },
+});
+
+// You can find the IDs of prices at https://pyth.network/developers/price-feed-ids
+const priceFeedId =
+  "0xff61491a931112ddf1bd8147cd1b641375f79f5825126d665480874634fd0ace"; // ETH/USD
+
+// Convert the price update to Starknet format.
+const priceUpdate = ByteBuffer.fromHex(currentPrices[0].vaa);
+
+// Query the fee required by Pyth to update the price on-chain.
+const fee = await pythContract.get_update_fee(
+  priceUpdate,
+  strkErc0Contract.address
+);
+
+// Approve fee withdrawal.
+strkErc0Contract.connect(account0);
+let tx = await strkErc0Contract.approve(pythContract.address, fee);
+await provider.waitForTransaction(tx.transaction_hash);
+
+// Submit a transaction to your contract using the price update data.
+pythContract.connect(account0);
+tx = await pythContract.update_price_feeds(priceUpdate);
+await provider.waitForTransaction(tx.transaction_hash);
```

Comment:
i think it's confusing that this example isn't interacting with the sample application above. Particularly confusing part is that this example is showing a different fee logic (the caller has to approve Pyth instead of the application contract).